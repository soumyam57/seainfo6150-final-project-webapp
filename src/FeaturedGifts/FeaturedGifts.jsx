import React from 'react';
import styles from "./FeaturedGifts.module.css";
import photoframe from "../images/photoframe.jpg";
import livetree from "../images/livetree.jpg";

const FeaturedGifts = () => {
    return (
        <div>
            <h1 className={styles.title}>
                Popular Today
            </h1>
            <div className={styles.container}>
                <div className={styles.grid}>
                    <div className={styles.div}>
                        <img className={styles.image} src={livetree} alt=""/>
                        <a className={styles.text} href="/festival/livetree">Christmas Reindeer Live Tree</a>
                    </div>
                    <div className={styles.div}>
                        <img className={styles.image} src={photoframe} alt=""/>
                        <a className={styles.text} href="/wedding/photoholder">Couple Gifts Photo Holder</a>
                    </div>
                </div>
            </div>
        </div>
    )
}


export default FeaturedGifts
