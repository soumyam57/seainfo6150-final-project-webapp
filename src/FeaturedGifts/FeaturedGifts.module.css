.container {
  box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5);    
  border: 0px dashed #666;
  display: grid;
  grid-template-columns: 1fr;
  justify-items: center;
  height: 90%;
}

@media screen and (min-width: 700px) {
  .container {
    grid-template-columns: repeat(1, 1fr);
  }
}

@media screen and (min-width: 800px) {
  .container {
    grid-template-columns: repeat(1, 1fr);
  }
}

.flex{
  flex-direction: column;
  flex-wrap: row wrap;
  padding: 2%;
  height: 100%;
  align-items: center;
}


.image{
  padding: 2%;
  border-radius: 10%;
  width: 46%;
  height: 80%
}

.title {
  background-color:  rgb(179, 148, 148);
  padding: 10px 50px;
  color:rgb(83, 15, 15);
  font-style: italic;
  font-weight: bold;
  text-transform: uppercase;
}