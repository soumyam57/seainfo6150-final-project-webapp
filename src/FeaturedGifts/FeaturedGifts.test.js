import React from "react";
import { render } from "@testing-library/react";
import FeaturedGifts from "./FeaturedGifts.jsx";
import { BrowserRouter } from 'react-router-dom';

it("renders the FeaturedGifts component correctly", () => {
  const { container } = render(
    <BrowserRouter>  
        <FeaturedGifts/>
    </BrowserRouter> 
  );
  expect(container).toMatchSnapshot();
});
