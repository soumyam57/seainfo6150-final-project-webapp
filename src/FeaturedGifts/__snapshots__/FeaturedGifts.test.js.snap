// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FeaturedGifts component correctly 1`] = `
<div>
  <div>
    <h1
      class="title"
    >
      Popular Today
    </h1>
    <div
      class="container"
    >
      <div
        class="grid"
      >
        <div
          class="div"
        >
          <img
            alt=""
            class="image"
            src="livetree.jpg"
          />
          <a
            class="text"
            href="/festival/livetree"
          >
            Christmas Reindeer Live Tree
          </a>
        </div>
        <div
          class="div"
        >
          <img
            alt=""
            class="image"
            src="photoframe.jpg"
          />
          <a
            class="text"
            href="/wedding/photoholder"
          >
            Couple Gifts Photo Holder
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
